Framework/WorkflowAlgorithms/src/RefRoi.cpp

```diff
@@ -165,19 +165,17 @@ void RefRoi::extract2D() {
   int integrated_axis_max = integrate_y ? ymax : xmax;
 
   for (int i = main_axis_min; i <= main_axis_max; i++) {
-    size_t output_index = i;
-    if (sum_pixels)
-      output_index = 0;
+    size_t output_index = (sum_pixels) ? 0 : i;
 
-    MantidVec &YOut = outputWS->dataY(output_index);
-    MantidVec &EOut = outputWS->dataE(output_index);
+    auto &YOut = outputWS->mutableY(output_index);
+    auto &EOut = outputWS->mutableE(output_index);
     MantidVec signal_vector(YOut.size(), 0.0);
     MantidVec error_vector(YOut.size(), 0.0);
 
     for (int j = integrated_axis_min; j <= integrated_axis_max; j++) {
       int index = integrate_y ? m_nYPixel * i + j : m_nYPixel * j + i;
-      const MantidVec &YIn = inputWS->readY(index);
-      const MantidVec &EIn = inputWS->readE(index);
+      const auto &YIn = inputWS->y(index);
+      const auto &EIn = inputWS->e(index);
 
```

Comment:
From a performance perspective the lines that read
`... += EIn[t_index] * EIn[t_index]` access the array twice - I would have thought that compilers were smart enough to optimize this out but at least MSVC on Release didn't.
Instead doing:
```c++
const double EInValue = EIn[t_index];
... += EInValue * EInValue;
```
Will increase the performance